Keda.Cosmosdb.Scaler/src/Services/ExternalScalerService.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Google.Protobuf.Collections;
+using Grpc.Core;
+using Keda.Cosmosdb.Scaler.Protos;
+using Microsoft.Azure.Documents.ChangeFeedProcessor;
+using Microsoft.Azure.Documents.Client;
+
+namespace Keda.Cosmosdb.Scaler.Services
+{
+    public class ExternalScalerService : ExternalScaler.ExternalScalerBase
+    {
+        private static ConsoleLogger logger = new ConsoleLogger();
+        public override async Task<IsActiveResponse> IsActive(ScaledObjectRef request, ServerCallContext context)
+        {
+            bool isActive = false;
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScalerMetadata);
+                var workToBeDone = await GetEstimatedWork(trigger);
+
+                isActive = workToBeDone > 0;
+
+                if (isActive)
+                {
+                    logger.Log(string.Format("Activating to 1 for cosmosDB database {0} collection {1}", trigger.DatabaseName, trigger.CollectionName));
+                }
+            }
+            catch(Exception ex)
+            {
+                logger.Log(ex.Message);
+            }
+
+            return new IsActiveResponse
+            {
+                Result = isActive
+            };
+        }
+
+        public override Task<GetMetricSpecResponse> GetMetricSpec(ScaledObjectRef request, ServerCallContext context)
+        {
+            var resp = new GetMetricSpecResponse();
+            resp.MetricSpecs.Add(new MetricSpec
+            {
+                MetricName = "WorkToBeDone",
+                TargetSize = 1
```

Comment:
We should read this from `request.ScalerMetadata`.  maybe `"targetWorkToBeDone"`